tools/riemann-aws/bin/riemann-s3-status

```diff
@@ -0,0 +1,101 @@
+  include Riemann::Tools
+  require 'fog'
+  require 'time'
+
+  opt :fog_credentials_file, "Fog credentials file", :type => String
+  opt :fog_credential, "Fog credentials to use", :type => String
+  opt :aws_access, "AWS Access Key", :type => String
+  opt :aws_secret, "AWS Secret Key", :type => String
+  opt :aws_region, "AWS Region", :type => String, :default => "eu-west-1"
+  opt :buckets, "Buckets to pull metrics from, multi=true", :type => String, :multi => true, :required => true
+  opt :statistic, "Statistic to retrieve, multi=true, e.g. --statistic=Average --statistic=Maximum", :type => String, :multi => true, :required => true
+
+
+  def base_metrics
+    # get last 60 seconds
+    start_time = (Time.now.utc - 3600 * 24 * 1).iso8601
+    end_time = Time.now.utc.iso8601
+
+    # The base query that all metrics would get
+    metric_base = {
+        "Namespace" => "AWS/S3",
+        "StartTime" => start_time,
+        "EndTime" => end_time,
+        "Period" => 3600,
+        "MetricName" => "NumberOfObjects",
+    }
+
+    metric_base
+  end
+
+
+  def tick
+    if options[:fog_credentials_file]
+      Fog.credentials_path = options[:fog_credentials_file]
+      Fog.credential = options[:fog_credential].to_sym
+      connection = Fog::AWS::CloudWatch.new
+    else
+      if options[:aws_access] && options[:aws_secret]
+        connection = Fog::AWS::CloudWatch.new({
+                                                  :aws_access_key_id => options[:aws_access],
+                                                  :aws_secret_access_key => options[:aws_secret],
+                                                  :region => options[:aws_region]
+                                              })
+      else
+        connection = Fog::AWS::CloudWatch.new({
+                                                  :use_iam_profile => true,
+                                                  :region => options[:aws_region]
+                                              })
+      end
+    end
+
+    options[:statistic].each do |statistic|
+      options[:buckets].each do |bucket|
+
+        metric_base_options = base_metrics
+        metric_base_options["Statistics"] = statistic
+        metric_base_options["Dimensions"] = [
+            {"Name" => "BucketName", "Value" => bucket},
+            {"Name" => "StorageType", "Value" => "AllStorageTypes"}]
+
+        # puts metric_base_options
+        result = connection.get_metric_statistics(metric_base_options)
+        # puts result
+        if result.body["GetMetricStatisticsResult"]["Datapoints"].empty?
+          next
+        end
+          result.body["GetMetricStatisticsResult"]["Datapoints"][0].keys.sort.each do |stat_type|
+          next if stat_type == "Unit"
+          next if stat_type == "Timestamp"
+
+          unit = result.body["GetMetricStatisticsResult"]["Datapoints"][0]["Unit"]
+          metric = result.body["GetMetricStatisticsResult"]["Datapoints"][0][stat_type]
+          puts metric
```

Comment:
Riemann S3 Status by asking Cloudwatch, since this is only updated once a day, the other utility is getting the bucket object count by listing and counting the contents. For that we need the upper limit to not do too much querying.